File: server/apps/nexotype/schemas/asset/therapeutic_peptide_schemas.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime

# ==========================================
# Request Schemas
# ==========================================

class TherapeuticPeptideCreate(BaseModel):
    """Schema for creating a new therapeutic peptide"""
    uid: str
    name: str
    project_code: str | None = None
    sequence_aa: str
    purity_grade: str | None = None


class TherapeuticPeptideUpdate(BaseModel):
    """Schema for updating a therapeutic peptide"""
    uid: str = None
    name: str = None
    project_code: str | None = None
    sequence_aa: str = None
    purity_grade: str | None = None


# ==========================================
# Response Schemas
# ==========================================

class TherapeuticPeptideDetail(BaseModel):
    """Schema for therapeutic peptide details"""
    id: int
    uid: str
    name: str
    project_code: str | None = None
    asset_type: str
    sequence_aa: str
    purity_grade: str | None = None
    created_at: datetime
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class TherapeuticPeptideListResponse(BaseModel):
    """Response schema for listing therapeutic peptides"""
    success: bool
    data: list[TherapeuticPeptideDetail] | None = None
    count: int = 0
    error: str | None = None


class TherapeuticPeptideResponse(BaseModel):
    """Response schema for single therapeutic peptide operations"""
    success: bool
    data: TherapeuticPeptideDetail | None = None
    error: str | None = None


class MessageResponse(BaseModel):
    """Simple message response"""
    success: bool
    message: str | None = None
    error: str | None = None

File: client/src/modules/nexotype/schemas/asset/therapeutic-peptide.schemas.ts
import { z } from 'zod';

// ==========================================
// Full Schema (GET responses)
// ==========================================

export const TherapeuticPeptideSchema = z.object({
  id: z.number(),
  uid: z.string().min(1).max(100),
  name: z.string().min(1).max(255),
  project_code: z.string().max(50).nullable().optional(),
  asset_type: z.string().max(50),
  sequence_aa: z.string().min(1),
  purity_grade: z.string().max(20).nullable().optional(),
  created_at: z.string(),
  updated_at: z.string().nullable(),
});

// ==========================================
// Input Schemas (POST/PUT)
// ==========================================

export const CreateTherapeuticPeptideSchema = z.object({
  uid: z.string().min(1).max(100),
  name: z.string().min(1).max(255),
  project_code: z.string().max(50).nullable().optional(),
  sequence_aa: z.string().min(1),
  purity_grade: z.string().max(20).nullable().optional(),
});

export const UpdateTherapeuticPeptideSchema = z.object({
  uid: z.string().min(1).max(100).optional(),
  name: z.string().min(1).max(255).optional(),
  project_code: z.string().max(50).nullable().optional(),
  sequence_aa: z.string().min(1).optional(),
  purity_grade: z.string().max(20).nullable().optional(),
});

// ==========================================
// Types
// ==========================================

export type TherapeuticPeptide = z.infer<typeof TherapeuticPeptideSchema>;
export type CreateTherapeuticPeptide = z.infer<typeof CreateTherapeuticPeptideSchema>;
export type UpdateTherapeuticPeptide = z.infer<typeof UpdateTherapeuticPeptideSchema>;

// ==========================================
// Response Types
// ==========================================

export type TherapeuticPeptideResponse = {
  success: boolean;
  data?: TherapeuticPeptide;
  error?: string;
};

export type TherapeuticPeptidesResponse = {
  success: boolean;
  data?: TherapeuticPeptide[];
  count?: number;
  error?: string;
};
